### PR TITLE
[AAE-534] Search - extract search text input from SearchControlComponent

### PR DESCRIPTION
**Pull request by SilviuCPopa**
**Please check if the PR fulfills these requirements**

> - [x] The commit message follows our [guidelines](https://github.com/Alfresco/alfresco-ng2-components/wiki/Commit-format)
> - [x] Tests for the changes have been added (for bug fixes / features)
> - [x] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x")

> - [ ] Bugfix
> - [x] Feature
> - [ ] Code style update (formatting, local variables)
> - [ ] Refactoring (no functional changes, no api changes)
> - [ ] Build related changes
> - [ ] Documentation
> - [ ] Other... Please describe:

**What is the current behaviour?** (You can also link to an open issue here)
https://issues.alfresco.com/jira/browse/AAE-534

**What is the new behaviour?**

**Does this PR introduce a breaking change?** (check one with "x")

> - [ ] Yes
> - [x] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

File: lib/core/search-text/search-text-input.component.ts [hunk 1]
```diff
@@ -0,0 +1,271 @@
...
+            this.toggleSearchBar();
+        }
+    }
+
+    isSearchBarActive() {
+        return this.subscriptAnimationState.value === 'active' && this.liveSearchEnabled;
+    }
+
+    ngOnDestroy() {
+
```
Comment (eromano) [inline lib/core/search-text/search-text-input.component.ts]: this space is not needed

File: lib/core/search-text/search-text-input.component.ts [hunk 2]
```diff
@@ -0,0 +1,271 @@
...
+    }
+
+    private getDefaultState(dir: string): any {
+        if (this.dir) {
+            return this.getAnimationState(dir);
+        }
+        return this.animationStates.ltr.inactive;
+    }
+
+    private getAnimationState(dir: string) {
```
Comment (eromano) [inline lib/core/search-text/search-text-input.component.ts]: missing return type

File: lib/content-services/src/lib/search/components/search-control.component.html [hunk 3]
```diff
@@ -1,46 +1,26 @@
...
-                   (blur)="onBlur($event)"
-                   (keyup.escape)="toggleSearchBar()"
-                   (keyup.arrowdown)="selectFirstResult()"
-                   (ngModelChange)="inputChange($event)"
-                   [searchAutocomplete]="auto"
-                   (keyup.enter)="searchSubmit($event)">
-        </mat-form-field>
-    </div>
-</div>
+<adf-search-text-input  #searchTextInput
```
Comment (eromano) [inline lib/content-services/src/lib/search/components/search-control.component.html]: there where an *ngIf="isLoggedIn()" before here

File: docs/core/components/search-text-input.component.md [hunk 4]
```diff
@@ -0,0 +1,48 @@
+---
+Title: Search Cloud Component
+Added: v3.5.0
```
Comment (eromano) [inline docs/core/components/search-text-input.component.md]: added should be 3.6.0 next version

File: docs/core/components/search-text-input.component.md [hunk 5]
```diff
@@ -0,0 +1,48 @@
+---
+Title: Search Cloud Component
```
Comment (eromano) [inline docs/core/components/search-text-input.component.md]: title is wrong

File: docs/core/components/search-text-input.component.md [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+Added: v3.5.0
+Status: Active
+Last reviewed: 2019-10-24
+---
+
+# [Search Text Input Component](../../../lib/core/search-text/search-text-input.component.ts "Defined in search-text-input.component.ts")
+
+Displays a input text that supports autocompletion
+
```
Comment (eromano) [inline docs/core/components/search-text-input.component.md]: is it possible to have a png of the component?

File: lib/core/search-text/search-text-input.component.ts [hunk 7]
```diff
@@ -0,0 +1,271 @@
...
+        this.toggleSearchBar();
+    }
+
+    activateToolbar() {
+        if (!this.isSearchBarActive()) {
+            this.toggleSearchBar();
+        }
+    }
+
+    isSearchBarActive() {
```
Comment (DenysVuika) [inline lib/core/search-text/search-text-input.component.ts]: return type missing

File: lib/core/search-text/search-trigger.directive.ts [hunk 8]
```diff
@@ -62,7 +61,7 @@ export class SearchTriggerDirective implements ControlValueAccessor, OnDestroy {
     private onDestroy$: Subject<boolean> = new Subject<boolean>();
 
     @Input('searchAutocomplete')
-    searchPanel: SearchComponent;
+    searchPanel: any;
```
Comment (DenysVuika) [inline lib/core/search-text/search-trigger.directive.ts]: why removing type?

File: lib/core/search-text/search-trigger.directive.ts [hunk 9]
```diff
@@ -186,7 +185,7 @@ export class SearchTriggerDirective implements ControlValueAccessor, OnDestroy {
     }
 
     private isNoResultOption() {
```
Comment (DenysVuika) [inline lib/core/search-text/search-trigger.directive.ts]: missing return type

**Pull request merged**